.github/workflows/flink-ci-workflow.yml

```diff
@@ -0,0 +1,77 @@
+#
+# Licensed to the Apache Software Foundation (ASF) under one
+# or more contributor license agreements.  See the NOTICE file
+# distributed with this work for additional information
+# regarding copyright ownership.  The ASF licenses this file
+# to you under the Apache License, Version 2.0 (the
+# "License"); you may not use this file except in compliance
+# with the License.  You may obtain a copy of the License at
+#
+#   http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing,
+# software distributed under the License is distributed on an
+# "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+# KIND, either express or implied.  See the License for the
+# specific language governing permissions and limitations
+# under the License.
+#
+
+
+# Workflow that is intended to be called from the main "CI Orchestrator"
+# We do this to ensure that:
+#  - all workflows that are launched by a PR will fail fast if one of them fails
+#  - all duplicate workflow runs are cancelled (i.e. only run workflows for the most recent push).
+name: "Flink CI Workflow"
+on:
+  workflow_call:
```

Comment:
Essentially I think if we update all of the current main CI workflows to use `on: workflow_call`, and then call them from the orchestrator, we can heavily parallelize and slice and dice tests into various files and still treat them like they were one big run.

We'd then have the choice of keeping logic either all in the current files, or moving some of it upwards if we desired.

One example reason to move some of it upwards would be to first perform the build in one job the other tests jobs depend on, so they all benefit from the same build cache. But that's a concern for further in the future. For now, just getting the tests to exit early when one fails would be a big win.